### DISCUSSION TITLE
How can I configure it to ensure the FPS is not affected by the mouse's movement?

### DISCUSSION THREAD

**Discussion by 247gzs** [emilk/egui#4956] [Q&A] [answered]
I have set ctx.request_repaint_after_for(Duration::from_secs(5), ctx.viewport_id()); in my code so that it updates once every 5 seconds when the mouse is not over the canvas. However, when the mouse moves over the canvas, the FPS significantly increases, which causes me a lot of trouble. How can I configure it to ensure the FPS is not affected by the mouse's movement?

egui： 0.28

**Comment by YgorSouza** [comment, marked as answer]
You can't, at least not with eframe. The app always repaints on user interaction. The `request_repaint` methods are just to ensure it keeps updating when the user isn't doing anything, if the UI might change for some other reason. There is #4880 which aims to skip the repaint if the hovered widget isn't interactive, but it hasn't been reviewed yet.

Generally you should assume that your UI code will be run at 60fps, because it will be whenever the user is interacting or an animation is in progress. If the problem is that the code is too expensive to run that often, you have to cache the results between frames. Egui provides the [cache](https://docs.rs/egui/0.28.1/egui/util/cache/index.html) module to help with that. You can also use the [time](https://docs.rs/egui/0.28.1/egui/struct.InputState.html#structfield.time) field obtainable via the [input](https://docs.rs/egui/0.28.1/egui/struct.Ui.html#method.input) to determine when to refresh things.

**Reply by 247gzs** [reply]
Thank you very much for your reply. I understand
